### PR TITLE
Add a new UI with webworker REPL support

### PR DESCRIPTION
**Pull request by katharosada**
I've kept it as minimal as I can, but for now it works ...mostly! Just sending this PR for feedback.

Notes:
* Copies the `python.js`, `python.wasm` and `python.data` files into the folder - happy to find a better way to do this (symlink?)
* Supports REPL only, I've not made it use the `<textarea>` yet so trying to click the 'Run' button will just error out.
* No matter what I tried, I cant get stdin to be auto-detected as a tty (apparently `isatty` always returns false in emscripten) but using the args `-i -` forces interactive mode.
* Not very performant on stdout/stderr since it's individually sending each character via `postMessage` lol. (It'd be better to buffer it and use postMessage only on flush).
* Very limited terminal control code handling, only supports typing visible characters and backspace (won't deal well with line wrapping)
* Doesn't handle copy/paste... at all.
* Attempts to encode 1 byte at a time into ASCII - understandably this completely breaks any non-ascii utf-8

### CODE REVIEW

File: browser-ui/index.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>wasm-python terminal</title>
+    <link rel="stylesheet" href="https://unpkg.com/xterm@4.16.0/css/xterm.css" crossorigin/>
+    <link rel="stylesheet" href="/styles.css"/>
```
Comment (tiran) [inline browser-ui/index.html]: Make import relative so we can deploy the shell in a sub directory on Ethan's GH page.
```suggestion
    <link rel="stylesheet" href="./styles.css"/>
```

File: browser-ui/index.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<body>
+    <textarea id="code"></textarea>
+    <div class="button-container">
+      <button id="run" disabled>Run</button>
+      <button id="repl" disabled>REPL</button>
+      <button id="clear" disabled>Clear</button>
+    </div>
+    <div id="terminal"></div>
+</body>
+</html>
```
Comment (tiran) [inline browser-ui/index.html]: Missing newline at EOF
```suggestion
</html>

```

File: browser-ui/server.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        self.send_my_headers()
+
+        server.SimpleHTTPRequestHandler.end_headers(self)
+
+    def send_my_headers(self):
+        self.send_header("Cross-Origin-Opener-Policy", "same-origin")
+        self.send_header("Cross-Origin-Embedder-Policy", "require-corp")
+
+
+with socketserver.TCPServer(("", PORT), MyHTTPRequestHandler) as httpd:
```
Comment (tiran) [inline browser-ui/server.py]: ``http.server`` uses ``ThreadingTCPServer``

You could also use ``http.server.test(HandlerClass=MyHTTPReqyestHandler, protocol="HTTP/1.1")``

**Review by tiran — COMMENTED** [review]
Fantastic! Thanks a lot.

I noticed that all files are missing a newline at end of file.

**Review by emmatyping — APPROVED** [review]
This looks fantastic! I don't have anything to add other than +1 to Christian's review comments. Very much looking forward to landing this :)

File: browser-ui/server.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+
+import socketserver
+from http import server # Python 3
+
+PORT = 8000
```
Comment (kumaraditya303) [inline browser-ui/server.py]:
```suggestion
from http import server # Python 3
import os

PORT = int(os.getenv("PORT", 8000))
```

File: browser-ui/server.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+class MyHTTPRequestHandler(server.SimpleHTTPRequestHandler):
+    def end_headers(self):
+        self.send_my_headers()
+
+        server.SimpleHTTPRequestHandler.end_headers(self)
```
Comment (kumaraditya303) [inline browser-ui/server.py]:
```suggestion
        super().end_headers()
```

File: browser-ui/server.py (hunk 4 above)
Comment (tiran) [reply browser-ui/server.py]: If we decide to make the port configurable, then we should use argparse instead. For now hard-coded port is good enough.

File: browser-ui/server.py (hunk 3 above)
Comment (katharosada) [reply browser-ui/server.py]: Happy to make the change, but um, what's the difference?

Comment (katharosada) [reply browser-ui/server.py]: Actually, never mind I get it now. :)

**Comment by katharosada** [comment]
Fixed up all the things! Even added `argparse` to specify the port for the http server.
Also, I removed the textarea and `Run` button since I didn't implement them yet. I can put that in a separate PR.

**Review by emmatyping — APPROVED** [review]

**Comment by emmatyping** [comment]
Thank you so much!

**Pull request merged**